core/src/test/java/io/confluent/rest/ConnectionDurationFilterTest.java

```diff
@@ -0,0 +1,225 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.collect.ImmutableMap;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriBuilder;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+
+/*
+ *  Note on client support --
+ *  The Apache HttpClient library v4.x does not support HTTP/2.0, so we use
+ *  OkHttp for HTTP/2.0 tests.
+ */
+public class ConnectionDurationFilterTest {
+
+  static Server server;
+  private static final long MAX_CONNECTION_DURATION_MS = 100;
+
+  @BeforeAll
+  public static void setUp() throws Exception {
+    MyAddressApplication application =
+        new MyAddressApplication(
+            new MyAddressConfig(
+                ImmutableMap.of(
+                    "listeners", "http://localhost:8080",
+                    "http2.enabled", "true",
+                    "max.connection.duration.ms", String.valueOf(MAX_CONNECTION_DURATION_MS)))
+        );
+    server = application.createServer();
+    server.start();
+  }
+
+  @AfterAll
+  public static void tearDown() throws Exception {
+    server.stop();
+  }
+
+  @Test
+  public void connectionDurationTest_ApacheHttpClient_Ephemeral_HTTP1() throws Exception {
+
+    CloseableHttpClient client = createPersistentClient();
+    List<String> addressList = new ArrayList<>();
+
+    // Send a few requests with a delay between consecutive requests that is longer
+    // than the connection duration threshold.
+    // Each request should succeed and the server should return a different address for each request.
+    for (int i=0; i<5; i++) {
+
+      HttpGet request = createRequest(server.getURI());
+      assertEquals("HTTP/1.1", request.getProtocolVersion().toString());
```

Comment:
why the request is http1.1 when your server has http2 enabled?

Reply:
That's a limitation of the Apache http client. 
HTTP client version 4.x only supports http 1.x.
HTTP 2.x support was added in version 5.x (see [here](https://stackoverflow.com/a/59566041)).

Using 5.x http client would require a major version upgrade of that package  -- there are a number of major changes (including relocated packages, etc) making 4.x -> 5.x migration non-trivial  (probably a separate future pull request on its own).

Reply:
I think we have jetty http client code example across this repo. You might want to take a look.